teuthology/misc.py

```diff
@@ -885,18 +885,24 @@ def get_scratch_devices(remote):
     return retval
 
 
-def wait_until_healthy(ctx, remote, ceph_cluster='ceph'):
+def wait_until_healthy(ctx, remote, ceph_cluster='ceph', use_sudo=False):
     """
     Wait until a Ceph cluster is healthy. Give up after 15min.
     """
     testdir = get_testdir(ctx)
+    # when cluster is setup using ceph-deploy or ansible
+    # access to admin key is readonly for ceph user
+    sudo = None
+    if use_sudo:
+        sudo = 'sudo'
     with safe_while(tries=(900 / 6), action="wait_until_healthy") as proceed:
         while proceed():
             r = remote.run(
                 args=[
                     'adjust-ulimits',
                     'ceph-coverage',
                     '{tdir}/archive/coverage'.format(tdir=testdir),
+                    '{sudo}'.format(sudo=sudo),
```

Comment:
i don't think this works if `sudo` is `None`. instead maybe we can rename the parameter of `use_sudo` to `sudo` and concat a `cmd` list to `args`, like:
```py
cmd = ['ceph', '--cluster', ceph_cluster, 'health']
if sudo:
  cmd.insert(0, 'sudo')
with safe_while(...):
  while proceed():
    r = remote.run(
      args=[....] + cmd,
      stdout=StringIO(),
      logger=log.getChild('health')
    )
```